estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from odoo import fields, models, api
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Property offer"
+
+    price = fields.Float(required=True)
+    partner_id = fields.Many2one("res.partner", required=True, string="Buyer")
+    status = fields.Selection(selection=[("awaiting", "Awaiting"), ("refused", "Refused"), ("accepted", "Accepted")], required=True, default="awaiting", copy=False)
+
+    property_id = fields.Many2one("estate.property", required=True)
+
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+
+    _sql_constraints = [
+        ("price", "CHECK(price > 0)", "Offer price must be positive.")
+    ]
+
+    @api.depends("validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            origin = record.create_date if record.create_date else fields.Date.today()
```

Comment:
Really feels redundant otherwise. Note that a standard if statement was also acceptable and arguably easier to read
```suggestion
            origin = record.create_date or fields.Date.today()
```